src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -1236,7 +1237,69 @@ private static bool SupportsSpellCheck(JToken token)
         return serverCapabilities?.SpellCheckingProvider ?? false;
     }
 
-    private async Task<TResult?> DelegateTextDocumentPositionRequestAsync<TResult>(DelegatedPositionParams request, string methodName, CancellationToken cancellationToken)
+    public override async Task<VSProjectContextList?> ProjectContextsAsync(DelegatedProjectContextsParams request, CancellationToken cancellationToken)
+    {
+        var hostDocument = request.HostDocument;
+        var (synchronized, virtualDocument) = await _documentSynchronizer.TrySynchronizeVirtualDocumentAsync<CSharpVirtualDocumentSnapshot>(
+            hostDocument.Version,
+            hostDocument.Uri,
+            cancellationToken).ConfigureAwait(false);
+
+        if (!synchronized)
+        {
+            return null;
+        }
+
+        var projectContextParams = new VSGetProjectContextsParams()
+        {
+            TextDocument = new TextDocumentItem()
+            {
+                LanguageId = CodeAnalysis.LanguageNames.CSharp,
+                Uri = virtualDocument.Uri,
+                Version = virtualDocument.Snapshot.Version.VersionNumber,
+                Text = virtualDocument.Snapshot.GetText(),
```

Comment:
WTF, the ProjectContext request sends the document contents?

What does Roslyn do with them?

Reply:
No idea... nor do I know why it was this instead of a TextDocumentIdentifier

Reply:
we likely could send empty